### PR TITLE
[MODULAR] Adds the option for missing a limb to the prosthetics menu

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Ports https://github.com/effigy-se/effigy-se/pull/267

Allows you to spawn with an amputated limb in exchange for 3 quirk points.
The way this works behind the scene is by making a new limb subtype that quickly detaches itself and qdels itself upon being applied. Character receives no mood penalty for losing this limb roundstart.

## How This Contributes To The Skyrat Roleplay Experience

- Same reason plasmaman limbs were - more open character design. Even if I imagine some goofiness will unfold thanks to someone not thinking their choices through.
- Someone asked why do they **have** to start with a prosthetic if they don't want it

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
noarm

</details>

## Changelog

:cl: LT3, unit0016
qol: You can now choose between limb, prosthetic, or no limb at all
/:cl:

### CODE REVIEW

**Comment by trazodont** [comment]
Finally I no longer have to cut my arm off in the QM office and have everyone storm in wondering who's being murdered round start.

**Review by tf-4 — COMMENTED** [review]
It should be restricted from sec roles

**Comment by trazodont** [comment]
> It should be restricted from sec roles

Genuinely curious. Why?

**Comment by lessthnthree** [comment]
> It should be restricted from sec roles

I'm fine with that.

**Comment by tf-4** [comment]
> > It should be restricted from sec roles
> 
> Genuinely curious. Why?

The same reason pacifist, or the wheelchair one is. It makes you too much of a liability for a role that relies on the support of the people in your department.
This isn't a problem in other roles, but for security it goes beyond the point of an RP hook and just generally tends to make the gameplay experience less enjoyable for the people trying to work with you.

**Comment by anconfuzedrock** [comment]
while I do want this, I would like to clarify the real reason there could be a problem with people starting with a missing limb- being impossible to handcuff. Having one arm isn't so bad usually (except in sec, which need two arms to fire an MCR etc), but it makes you a pain in the ass if you decide to cause trouble.

Does this give people with  1 or no legs a wheelchair?

File: modular_skyrat/modules/customization/modules/jobs/_job.dm [hunk 1]
```diff
@@ -13,11 +13,9 @@
 	var/list/species_blacklist
 	/// Which languages does the job require, associative to LANGUAGE_UNDERSTOOD or LANGUAGE_SPOKEN
 	var/list/required_languages = list(/datum/language/common = LANGUAGE_SPOKEN)
-
 	///Is this job veteran only? If so, then this job requires the player to be in the veteran_players.txt
 	var/veteran_only = FALSE
 
-
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/jobs/_job.dm]: Why are you removing empty lines in an unrelated file? No need to touch a file that's unrelated like this, just undo it so it doesn't add a weird bit to the git history of this file if there's no need to.

File: modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+/obj/item/bodypart/leg/left/selfdestruct/try_attach_limb(mob/living/carbon/limb_owner, special)
+	. = ..()
+	drop_limb()
+	qdel(src)
+
+/obj/item/bodypart/leg/right/selfdestruct/try_attach_limb(mob/living/carbon/limb_owner, special)
+	. = ..()
+	drop_limb()
+	qdel(src)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm]: My only concern is, what happens in the prefs menu? We try to avoid deleting stuff when in the prefs menu, and only doing visual changes as needed, without actually deleting the underlying things. I'm just worried that this doesn't follow what the rest of it might be following. Otherwise, sounds good to me.

Comment (GoldenAlpharex) [reply modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm]:
```suggestion
/// Self Destructing Bodyparts, For Augmentation. I'm leaving out heads + chests as, while it would be cool for synths, I also don't want people to start the round unrevivable sans botany because they're dumb as heck. You know who and what you are.
/obj/item/bodypart/arm/left/self_destruct/try_attach_limb(mob/living/carbon/limb_owner, special)
	. = ..()
	drop_limb()
	qdel(src)

/obj/item/bodypart/arm/right/self_destruct/try_attach_limb(mob/living/carbon/limb_owner, special)
	. = ..()
	drop_limb()
	qdel(src)

/obj/item/bodypart/leg/left/self_destruct/try_attach_limb(mob/living/carbon/limb_owner, special)
	. = ..()
	drop_limb()
	qdel(src)

/obj/item/bodypart/leg/right/self_destruct/try_attach_limb(mob/living/carbon/limb_owner, special)
	. = ..()
	drop_limb()
	qdel(src)
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm (hunk 2 above)
Comment (lessthnthree) [reply modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm]: > My only concern is, what happens in the prefs menu? We try to avoid deleting stuff when in the prefs menu, and only doing visual changes as needed, without actually deleting the underlying things. I'm just worried that this doesn't follow what the rest of it might be following. Otherwise, sounds good to me.

Renders fine in the preferences menu.

Uploading dreamseeker_tXx9zDLNLN.mp4…

Comment (lessthnthree) [reply modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm]: > My only concern is, what happens in the prefs menu? We try to avoid deleting stuff when in the prefs menu, and only doing visual changes as needed, without actually deleting the underlying things. I'm just worried that this doesn't follow what the rest of it might be following. Otherwise, sounds good to me.

Renders fine in the preferences menu.

https://github.com/Skyrat-SS13/Skyrat-tg/assets/83487515/ed692c4e-ead6-4eaa-873b-f3999407fcb0

Comment (Jfrost6) [reply modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm]: How does it work with markings? Does it save them?
[I need this feature so bad]

Comment (lessthnthree) [reply modular_skyrat/modules/customization/modules/surgery/bodyparts/parts.dm]: It remains in your prefs (and visible) but you don't spawn with the marking as your limb is gone.

**Comment by vinylspiders** [comment]
This looks good to me but maybe you may want to give people who spawn in with no legs a wheelchair or some means of getting around?

I can foresee people getting stuck that way and not being able to make it to a cryo pod to fix their character.

**Comment by GoldenAlpharex** [comment]
Also, where did the security restrictions go, anyway?

I'd be fine for it to allow someone to still play security just missing one arm, but quite obviously, missing two arms makes it just simply unplayable, just like any command role, so I'd rather avoid that.

I'm not sure if it's wise to have the one where they're missing a leg at roundstart be allowed for stuff like security/command either, because they're basically doomed to never wearing any shoes, and that sounds like it could cause issues. If you want to play a person with a crippled leg in Command at roundstart, just have a cheap prosthetic. There's no real reason why you wouldn't get one, even if a cheap one from Central just so you're still functional.

**Comment by lessthnthree** [comment]
We don't presently have code that can restrict you from a job slot based on quirks or limb count and my first stab at it I hit a wall, so it hasn't been a priority.

**Comment by Jfrost6** [comment]
Could this just be an IC issue maybe? If someone tries to play security with no arms then they can either get prosthetics put on or get fired?

**Comment by lessthnthree** [comment]
Made the static icon proc an early return so no icon is applied.

**Review by GoldenAlpharex — APPROVED** [review]
Alright, looking good and ready for a test-merge!

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'd still like for there to be a system that can handle restricting the players from joining certain jobs (i.e. Command and Security) based on if they're missing limbs from this.

**Comment by trazodont** [comment]
> I'd still like for there to be a system that can handle restricting the players from joining certain jobs (i.e. Command and Security) based on if they're missing limbs from this.

Genuine question, *how* many limbs are allowed to be missing before it's an issue? I play a 1 armed QM and would be so sad if this restricts them.

**Comment by lessthnthree** [comment]
> I'd still like for there to be a system that can handle restricting the players from joining certain jobs (i.e. Command and Security) based on if they're missing limbs from this.

I'm fine with leaving this TM until we can get this part added.

**Comment by Jfrost6** [comment]
Inhand loadout items get deleted along with the left arm, or don't even spawn, not sure. Tested to make sure on the same round and I only have the cane if I have a left arm.

**Comment by tf-4** [comment]
> We don't presently have code that can restrict you from a job slot based on quirks or limb count and my first stab at it I hit a wall, so it hasn't been a priority.

Yes we do, plenty of quirks have job restrictions

**Comment by lessthnthree** [comment]
This isn't a quirk, it's an augment.

**Comment by lessthnthree** [comment]
> > We don't presently have code that can restrict you from a job slot based on quirks or limb count and my first stab at it I hit a wall, so it hasn't been a priority.
> 
> Yes we do, plenty of quirks have job restrictions

This isn't a quirk, it's an augment.

**Comment by lessthnthree** [comment]
> > I'd still like for there to be a system that can handle restricting the players from joining certain jobs (i.e. Command and Security) based on if they're missing limbs from this.
> 
> Genuine question, _how_ many limbs are allowed to be missing before it's an issue? I play a 1 armed QM and would be so sad if this restricts them.

The restriction is for security only, as is the other restrictions based on limb related quirks.

**Comment by Iamgoofball** [comment]
Do we still need these restrictions? They hamper roleplay honestly.

**Comment by lessthnthree** [comment]
> Do we still need these restrictions? They hamper roleplay honestly.

Up to the head maintainers, who requested it added.

File: modular_skyrat/modules/customization/modules/jobs/_job.dm [hunk 3]
```diff
@@ -41,6 +43,16 @@
...
 	return FALSE
 
+/datum/job/proc/has_banned_augment(datum/preferences/pref)
+	if(!pref)
+		return FALSE
+	if(banned_augments)
+		var/list/player_augments = pref.augments
+		for(var/key in player_augments)
+			if(player_augments[key] in banned_augments)
+				return TRUE
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/jobs/_job.dm]:
```suggestion

	if(!banned_augments)
		return FALSE

	var/list/player_augments = pref.augments
	for(var/key in player_augments)
		if(player_augments[key] in banned_augments)
			return TRUE

```

File: modular_skyrat/modules/customization/modules/jobs/_job.dm [hunk 4]
```diff
@@ -75,6 +92,7 @@
 
 /datum/job/head_of_security
 	banned_quirks = list(SEC_RESTRICTED_QUIRKS, HEAD_RESTRICTED_QUIRKS)
+	banned_augments = list(SEC_RESTRICTED_AUGMENTS)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/jobs/_job.dm:101]: I'd like for these to be applied to command as well, mainly because if someone wants to be in command, I'd like to think they'd be provided at the very least a shitty prosthetic to make them at least somewhat capable of using all their limbs.

**Review by GoldenAlpharex — DISMISSED** [review]

File: modular_skyrat/modules/customization/modules/jobs/_job.dm (hunk 4 above)
Comment (lessthnthree) [reply modular_skyrat/modules/customization/modules/jobs/_job.dm:101]: Sure

**Review by vinylspiders — APPROVED** [review]
